.github/workflows/verifyHybridApp.yml

```diff
@@ -0,0 +1,141 @@
+name: Verify HybridApp build
+
+on:
+  workflow_call:
+  pull_request:
+    types: [opened, synchronize]
+    branches-ignore: [staging, production]
+    paths:
+      - '**.kt'
+      - '**.java'
+      - '**.swift'
+      - '**.mm'
+      - '**.h'
+      - '**.cpp'
+      - 'package.json'
+      - 'patches/**'
+      - 'android/build.gradle'
+      - 'android/AndroidManifest.xml'
+      - 'ios/Podfile.lock'
+      - 'ios/project.pbxproj'
+
+concurrency:
+  group: ${{ github.ref == 'refs/heads/main' && format('{0}-{1}', github.ref, github.sha) || github.ref }}-verify-main
+  cancel-in-progress: true
+
+jobs:
+  verify_android:
+    name: Verify Android HybridApp builds on main
+    runs-on: ubuntu-latest-xl
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+          ref: ${{ github.event.pull_request.head.sha }}
+          token: ${{ secrets.OS_BOTIFY_TOKEN }}
+          # fetch-depth: 0 is required in order to fetch the correct submodule branch
```

Comment:
This is very slow, and most likely not necessary (though a manual git fetch may be necessary in a later step). To start, what happens if we just remove this?

Also, is "the correct submodule branch" just main in Mobile-Expensify?

Reply:
> Also, is "the correct submodule branch" just main in Mobile-Expensify?

It's a specific Mobile-Expensify commit that is locked to the latest commit as/when each App deploy is done.

Reply:
Is that the case currently, or will it only be the case once https://github.com/Expensify/App/issues/55403 is completed?

Reply:
> My understanding is that currently all staging and production E/App builds use Mobile-Expensify main, but that there's consensus that this is a problem we need to fix. Am I mistaken?

Yeah, on each App staging build we bump the submodule to the latest commit (see every other commit [here](https://github.com/Expensify/App/commits/main/?author=OSBotify)).

The problem is that this leads to Mobile-Expensify changes that can break an App release candidate late in a checklist that can break the build without us realising. To solve that we plan to lock the Mobile-Expensify commit for the full checklist -- but we'll give ourselves the option to override this if we need to.